### PR TITLE
UserService is now fully functional.

### PR DESCRIPTION
**Pull request by Tschanzt**
This time the branches should be correct.

### CODE REVIEW

**Comment by Tschanzt** [comment]
@imscaradh the tests now pass. They were order dependent. ahhhh

File: src/main/java/ch/bfh/srs/srv/service/RoleService.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Created by tschanzt on 29/11/15.
+ */
+public class RoleService extends BaseService<Role>{
+
+    public RoleService(EntityManager em) {
```
Comment (imscaradh) [inline src/main/java/ch/bfh/srs/srv/service/RoleService.java]: The constructor should be protected :construction:

File: src/test/java/ch/bfh/srs/srv/service/RoleServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
+package ch.bfh.srs.srv.service;
```
Comment (imscaradh) [inline src/test/java/ch/bfh/srs/srv/service/RoleServiceTest.java]: Please add the copyright header

File: src/test/java/ch/bfh/srs/srv/service/RoleServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+import org.junit.Test;
+
+import java.util.List;
+
+import static junit.framework.Assert.assertNull;
+import static junit.framework.TestCase.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Created by tschanzt on 07/12/15.
```
Comment (imscaradh) [inline src/test/java/ch/bfh/srs/srv/service/RoleServiceTest.java]: Please remove this

File: src/test/java/ch/bfh/srs/srv/service/UserServiceTest.java [hunk 4]
```diff
@@ -92,33 +95,49 @@ public void testGetUser() {
...
         assertEquals("max@mustermann.ch", userEntity.getMail());
         assertEquals("Max", userEntity.getSurname());
         assertEquals("Mustermann", userEntity.getLastname());
-        assertEquals("asdf", userEntity.getPassword());
+        org.junit.Assert.assertNotNull("pw should not be null", userEntity.getPassword());
+    }
+
+    @Test
+    public void testHash() {
+        String pw = "fuckoff";
```
Comment (imscaradh) [inline src/test/java/ch/bfh/srs/srv/service/UserServiceTest.java]: Can we change the displeasing password?

**Comment by Tschanzt** [comment]
@imscaradh updated.

**Comment by imscaradh** [comment]
giphy

**Pull request merged**